.github/workflows/R-CMD-check.yaml

```
@@ -42,10 +47,22 @@ jobs:
           use-public-rspm: true
 
       - if: runner.os == 'Linux'
-        run: sudo apt-get install -y protobuf-compiler libprotobuf-dev libprotoc-dev
+        run: sudo apt-get install -y protobuf-compiler libprotobuf-dev libprotoc-dev cmake libcurl4-openssl-dev libssl-dev
 
       - uses: r-lib/actions/setup-r-dependencies@v1
         with:
           extra-packages: rcmdcheck
+          path: substrait-r
+
+      - name: Checkout Arrow repo
+        uses: actions/checkout@v2
+        with:
+          repository: apache/arrow
+          path: arrow
+
```

```suggestion
- name: Cache libarrow
  id: cache
  uses: actions/cache@v3
  with:
    path: |
      ~/install_dir 
    key: ${{ runner.os }}-${{ hashFiles('arrow/cpp/**') }}
```
This should cache arrow